### PR TITLE
Gold early access streaming.

### PR DESCRIPTION
**Pull request by Vindexus**
Requires this branch for your backend: https://github.com/monstercat/connect-backend/pull/3

### CODE REVIEW

File: src/js/deon-main.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-var endhost         = 'https://connect.monstercat.com'
+var endhost         = 'http://localhost:7071'
```
Comment (GiovanniMCMXCIX) [inline src/js/deon-main.js]: umm isn't this supposed to be https://connect.monstercat.com ?

Comment (tmathews) [inline src/js/deon-main.js]: RIP @Vindexus

File: src/js/deon-main.js [hunk 2]
```diff
@@ -546,6 +575,61 @@ function transformGoldSubscription (obj) {
...
+  }, {
+    id: 'early-streaming',
+    title: 'Early Streaming Access',
+    description: 'Listen to releases on Monstercat.com 24 hours before they are released to everyone else.',
+    image: 'http://i.imgur.com/677JvWv.jpg'
+  }, {
+    id: 'support-the-artists',
+    title: 'Support the Artists',
+    description: 'Artists are paid out from Gold subscriptions based on how much people listen to their songs.',
+    image: 'http://i.imgur.com/AQrsFqP.png'
```
Comment (GiovanniMCMXCIX) [inline src/js/deon-main.js]: also you forgot to change the stock footage to one more professional or something

Comment (tmathews) [inline src/js/deon-main.js]: @Vindexus does this contain gold page changes? The PR title is misleading. We need to wait until @juanferreras goes through and makes his changes before we do a pull into master.

**Comment by Vindexus** [comment]
@tmathews Sorry I should've just done this as a compare link instead of a pull request. I'm still waiting on art from our team and for the backend changes before this can actually be merged.

Could you look at the rest of the code anyways? I'd rather not wait for the art and the gold landing style changes.

File: src/css/deon-style.css [hunk 3]
```diff
@@ -3641,4 +3693,45 @@ body.bg {
...
+.countdown {
+  font-size: 1.3em;
+}
+
+ul.gold-perks a {
+  text-decoration: none;
+}
+
+section.gold-feature-section {
+  padding-bottom: 0em;
```
Comment (tmathews) [inline src/css/deon-style.css]: Maybe look for a class that already does this for you.

File: src/js/deon-main.js [hunk 4]
```diff
@@ -585,21 +669,41 @@ function scrollToAnimated (el, opts) {
...
 function scrollToHighlightHash () {
   if(location.hash) {
-    var el = document.querySelector(location.hash)
-    if(el) {
-      setTimeout(function () {
-        scrollToAnimated(el)
-        el.classList.add('anchor-highlight')
+    var attempts = 0
+    var attempt = function () {
+      console.log('attempts', attempts)
```
Comment (tmathews) [inline src/js/deon-main.js]: Chop log.

File: src/js/deon-main.js [hunk 5]
```diff
@@ -585,21 +669,41 @@ function scrollToAnimated (el, opts) {
...
+      var el = document.querySelector(location.hash)
+      if(el) {
+        console.log('foundya', el)
```
Comment (tmathews) [inline src/js/deon-main.js]: Chop all the logs.

File: src/js/deon-main.js [hunk 6]
```diff
@@ -677,6 +782,18 @@ function appendSongMetaData (tracks) {
...
+function completedHome (source, obj) {
+  startCountdownTicks()
+}
+
+function featuredReleaseCountdownEnd () {
+  loadSubSources(document.querySelector('[role="home-featured"]'), true)
+}
+
+function releasePageCountdownEnd () {
+  reloadPage()
```
Comment (tmathews) [inline src/js/deon-main.js]: Just call this method directly? Could be future proofing here, but I don't see it.

File: src/js/early-access.js [hunk 7]
```diff
@@ -0,0 +1,97 @@
+function zeroPad (num, size) {
+  var str = num.toString()
+  return '0'.repeat(size).substr(0,size - str.length)+str
+}
+
+function dateToCountdownString (date) {
+  var end = new Date(date)
+  var diff = dateToRemainingTime(date)
+  var cd = new Date(diff)
+  //console.log('end', end.toISOString())
```
Comment (tmathews) [inline src/js/early-access.js]: Growing a forest here.

File: src/js/early-access.js [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+  var westCoastUTCOffset = -8
+  var mwc = dateToMidnight(date, westCoastUTCOffset)
+  return mwc
+}
+
+function dateToMidnight (date, utcOffsetHours) {
+  return dateToTimeOffset(date, utcOffsetHours, 0, 0)
+}
+
+function startCountdownTicks () {
```
Comment (tmathews) [inline src/js/early-access.js]: Contains a never ending timer. You need to make sure you exit when no longer needed.

File: src/js/helpers.js [hunk 9]
```diff
@@ -261,6 +261,10 @@ var animatedScrollTo = function (element, to, duration, callback) {
     requestAnimFrame(animateScroll);
 };
 
+function randomChooser(n){
```
Comment (tmathews) [inline src/js/helpers.js]: What does this name mean?

File: src/js/declare.js [hunk 10]
```diff
@@ -302,6 +302,13 @@ function openRoute (target, container, matches) {
       }
     }
   }
+  var fns = openRoute.started
+  if (fns instanceof Array) {
```
Comment (tmathews) [inline src/js/declare.js]: Looks like duplicated code here. You can create a function do this little shindig for ya.

File: src/js/deon-main.js [hunk 11]
```diff
@@ -117,13 +117,16 @@ function getSession (done) {
 }
 
 function getSessionName () {
-  var names = [session.user.name, session.user.realName, session.user.email.substr(0, session.user.email.indexOf('@'))]
+  var names = []
+  if(session.user) {
+    names = names.concat([session.user.name, session.user.realName, session.user.email.substr(0, session.user.email.indexOf('@'))])
```
Comment (tmathews) [inline src/js/deon-main.js]: New lines are your friend.

File: src/js/deon-main.js [hunk 12]
```diff
@@ -606,13 +609,13 @@ function transformGoldLanding (obj) {
     id: 'discord',
     title: 'Gold-only Discord Chat',
     description: 'Come chat with us and other superfans in our Discord server.',
-    image: 'https://puu.sh/rQYda/ba737d1553.png',
+    image: 'https://puu.sh/rZ01O/a29b5d87a3.png',//https://puu.sh/rZ05a/74f78f4a63.png
```
Comment (tmathews) [inline src/js/deon-main.js]: Are these temporary? Do not use external services.

**Pull request merged**